packages/next/build/utils.ts

```diff
@@ -648,9 +648,12 @@ export async function buildStaticPaths(
             } in getStaticPaths for ${page}`
           )
         }
-
+        let replaced = `[${repeat ? '...' : ''}${validParamKey}]`
+        if (optional) {
+          replaced = `[${replaced}]`
```

Comment:
@Timer having `optional` on the regexp group makes this quite straightforward, but this should be possible as well by constructing a regex.